sql1.txt

```diff
@@ -6,38 +6,64 @@ SELECT * FROM Products LIMIT 5;
 1. Use an SQL SELECT statement to retrieve the first 10 rows of the Customer table, ordered by CustomerName.
 Paste your SQL statement below:
 
+SELECT * 
+FROM Customers 
+ORDER BY CustomerName 
+LIMIT 10;
 
 2. Use an SQL Select statement to retrieve the names and prices for all products that cost less than $20.
 Paste your SQL statement below:
 
+SELECT ProductName, Price 
+FROM Products 
+WHERE Price < 20;
 
 3. Retrieve all employees whose last name starts with C.  Paste your SQL statement below:
 
+SELECT * FROM Employees 
+WHERE LastName LIKE 'C%';
 
 4. Retrieve the order ID and customer name for all orders where the customer name starts with A.
 Here you will have to do a join of the Orders table and the Customers table.  Paste your SQL
 statement below.
 
+SELECT Orders.OrderId, Customers.CustomerName 
+FROM Orders 
+JOIN Customers ON Orders.CustomerID=Customers.CustomerID
+WHERE Customers.CustomerName 
+LIKE 'A%';
 
 5. Retrieve the list of customers, ordered by customer name, along with their order IDs.  Note
 that not every customer has an order, but be sure that you include those customers without orders
 in the results of the query.  Paste your SQL statement below.
 
-
+SELECT Customers.CustomerName, Orders.OrderID 
+FROM Customers 
+LEFT JOIN Orders ON Customers.CustomerID = Orders.CustomerID 
+ORDER BY Customers.CustomerName;
 
 6. Retrieve the list of customer names for customers that have no orders.  There are two ways to do this
 One is to use a subquery, as follows:
 
-SELECT CustomerName FROM Customers WHERE CustomerID NOT IN (SELECT CustomerID FROM ORDERS);
+SELECT CustomerName
+FROM Customers 
+WHERE CustomerID NOT IN (SELECT CustomerID FROM ORDERS);
 
 This is a subquery -- we haven't talked about those.  But, you can do the same thing with a left
 join by comparing the OrderID column with NULL.  Paste your SQL statement below.
 
-
+SELECT CustomerName
+FROM Customers
+LEFT JOIN Orders ON Customers.CustomerID = Orders.CustomerID
+WHERE Orders.OrderID IS NULL; 
 
 7. Retrieve the OrderID and ProductName of every order for a product with Tofu in the name.
 Paste your SQL statement below.  You will have to join Orders, OrderDetails, and Products.
 
-
-
+SELECT Orders.OrderID, Products.ProductName 
+FROM Orders 
+JOIN OrderDetails ON Orders.OrderID = OrderDetails.OrderID 
+JOIN Products ON OrderDetails.ProductID = Products.ProductID 
+WHERE Products.ProductName 
+LIKE '%Tofu%';
```

Comment:
Did you know you can [alias](https://www.w3schools.com/sql/sql_alias.asp) table names to make it easier to type things with `JOIN` statements? You can either use the `AS` keyword or simply type the alias directly following the table name.

You can also skip specifying the table for a column (`Product.ProductName`) if it is the only table with that column name (`ProductName`)

```suggestion
SELECT o.OrderID, ProductName 
FROM Orders o
JOIN OrderDetails d ON o.OrderID = d.OrderID 
JOIN Products p ON d.ProductID = p.ProductID 
WHERE ProductName 
LIKE '%Tofu%';
```

Reply:
Thank you! I used this at new assignment.
I will refactor my assignment.